### PR TITLE
[FIX] mail: fix `assertMailNotifications` when called in batch

### PR DESCRIPTION
**Pull request by std-odoo**
Bug
===
When we called assertMailNotifications in batch, with different
notification status, it might raise an error when it shouldn't.

The reason for that is we check that no mail at all are created, for
the entire batch instead of filtering on the related partner.

`assertNoMail` should only check if no mail are created for the partner
it receive in arguments.

Task-2782150

### CODE REVIEW

File: addons/mail/tests/common.py [hunk 1]
```diff
@@ -416,11 +416,25 @@ def assertNoMail(self, recipients, mail_message=None, author=None):
...
+    def assertNotSentEmail(self, recipients=None):
+        """Check no email was generated during gateway mock.
+
+        :param recipients: List of partner for which we will check that no email have been sent
+            If None, we will check that no email at all have been sent
+        """
+        if recipients is None:
+            mails = self._mails
+        else:
+            all_emails = recipients.mapped('email')
```
Comment (tde-banana-odoo) [inline addons/mail/tests/common.py:430]: Note that recipients could be a list of emails (a bit hybrid, as sometimes we have partners, sometimes only emails), a bit like what is done in assertEmailSent
```
all_emails = [
    email_to.email if isinstance(email_to, self.env['res.partner'].__class__) else email_to
    for email_to in recipients
]
```

**Review by tde-banana-odoo — COMMENTED** [review]
Hybrids validation !

**Comment by std-odoo** [comment]
@tde-banana-odoo Updated in both 14 and 15 :)

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**